### ISSUE TITLE
Undocumented feature: all fields expands to FieldResult

### ISSUE THREAD

**Issue by IcanDivideBy0** [async-graphql/async-graphql#47]
Hello,
I'm not sure if this is realted to the recent changes on `FieldResult` but I've just noticed that we can use `?` in field resolvers even if they don't return a `FieldResult`:

```rust
#[async_graphql::Object]
impl QueryRoot {
    #[field]
    async fn user(&self, ctx: &Context<'_>, id: ID) -> models::User {
        let id = ObjectId::with_string(&id.to_string())?;
        ctx.data::<Loaders>().user.load(id).await?
    }
}
```
It feel a bit odd, but is also very convenient and makes error management easier. The documentation doesn't seems to mention this at the moment, is this a feature we can rely on or is it subject to change in the future ?

Cheers

**Comment by sunli829** [comment]
The code generator modifies the function signature to add FieldResult to all return values, which is why this code works.
But I think this is a bug that should cause the compiler to report an error.

**Comment by sunli829** [comment]
I fixed this problem in version `1.10.6`, and now the compiler is reporting errors correctly.😁

**Comment by IcanDivideBy0** [comment]
Nice!
